### PR TITLE
Add scenario tests for deleting a registered application

### PR DESCRIPTION
**Pull request by ruwiniwj**
## Purpose
Add scenario tests for deleting a registered application(4.1.3.1, 4.1.3.2, 4.1.3.3, 4.1.3.4, 4.1.3.5)

### CODE REVIEW

File: product-scenarios/4-application-development-using-apis/4.1-application-developers-should-be-able-to-use-apis-in-their-applications/4.1.3-delete-registered-application/src/test/java/org/wso2/am/scenario/tests/delete/registered/application/DeleteRegisteredApplicationNegativeTestCase.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+public class DeleteRegisteredApplicationNegativeTestCase extends ScenarioTestBase {
+    private APIStoreRestClient apiStore;
+    private static final Log log = LogFactory.getLog(DeleteRegisteredApplicationNegativeTestCase.class);
+    private static final String ADMIN_LOGIN_USERNAME = "admin";
+    private static final String ADMIN_LOGIN_PW = "admin";
+    private static final String APPLICATION_NAME = "DeleteUnownedApplication";
+    private static final String SUBSCRIBER_USERNAME = "deleteAppSubscriber1";
+    private static final String SUBSCRIBER_PW = "deleteAppSubscriber1";
+    private static final String SUBSCRIBER2_USERNAME = "deleteAppSubscriber2";
```
Comment (rmsamitha) [inline product-scenarios/4-application-development-using-apis/4.1-application-developers-should-be-able-to-use-apis-in-their-applications/4.1.3-delete-registered-application/src/test/java/org/wso2/am/scenario/tests/delete/registered/application/DeleteRegisteredApplicationNegativeTestCase.java]: Why do we need these constants as static?

Comment (ruwiniwj) [reply product-scenarios/4-application-development-using-apis/4.1-application-developers-should-be-able-to-use-apis-in-their-applications/4.1.3-delete-registered-application/src/test/java/org/wso2/am/scenario/tests/delete/registered/application/DeleteRegisteredApplicationNegativeTestCase.java]: Removed static constants

File: product-scenarios/4-application-development-using-apis/4.1-application-developers-should-be-able-to-use-apis-in-their-applications/4.1.3-delete-registered-application/4.1.3-TESTCASES.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# 4.1.3 Test cases for deleting a registered application
+
```
Comment (praminda) [inline product-scenarios/4-application-development-using-apis/4.1-application-developers-should-be-able-to-use-apis-in-their-applications/4.1.3-delete-registered-application/4.1.3-TESTCASES.md]:
```suggestion
```

File: product-scenarios/4-application-development-using-apis/4.1-application-developers-should-be-able-to-use-apis-in-their-applications/4.1.3-delete-registered-application/pom.xml [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+    <dependencies>
+        <dependency>
+            <groupId>org.wso2.am</groupId>
+            <artifactId>org.wso2.am.integration.admin.clients</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.wso2.am</groupId>
+            <artifactId>scenarios-common</artifactId>
+            <version>2.6.0</version>
```
Comment (ruks) [inline product-scenarios/4-application-development-using-apis/4.1-application-developers-should-be-able-to-use-apis-in-their-applications/4.1.3-delete-registered-application/pom.xml]: any reason to define version here? Should it go to parent pom?

Comment (ruwiniwj) [reply product-scenarios/4-application-development-using-apis/4.1-application-developers-should-be-able-to-use-apis-in-their-applications/4.1.3-delete-registered-application/pom.xml]: Fixed with https://github.com/wso2/product-apim/pull/5210

File: product-scenarios/4-application-development-using-apis/4.1-application-developers-should-be-able-to-use-apis-in-their-applications/4.1.3-delete-registered-application/src/test/java/org/wso2/am/scenario/tests/delete/registered/application/DeleteRegisteredApplicationTestCase.java [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+        for (String name : applicationsList) {
+            apiStore.removeApplication(name);
+        }
+        for (String name : apiList) {
+            apiPublisher.deleteAPI(name, API_VERSION, CREATOR_PUBLISHER_USERNAME);
+        }
+        deleteUser(CREATOR_PUBLISHER_USERNAME, ADMIN_LOGIN_USERNAME, ADMIN_LOGIN_PW);
+        deleteUser(SUBSCRIBER_USERNAME, ADMIN_LOGIN_USERNAME, ADMIN_LOGIN_PW);
+    }
+}
```
Comment (nalakas) [inline product-scenarios/4-application-development-using-apis/4.1-application-developers-should-be-able-to-use-apis-in-their-applications/4.1.3-delete-registered-application/src/test/java/org/wso2/am/scenario/tests/delete/registered/application/DeleteRegisteredApplicationTestCase.java]: Add new line

File: product-scenarios/4-application-development-using-apis/4.1-application-developers-should-be-able-to-use-apis-in-their-applications/4.1.3-delete-registered-application/src/test/java/org/wso2/am/scenario/tests/delete/registered/application/DeleteRegisteredApplicationNegativeTestCase.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
+package org.wso2.am.scenario.tests.delete.registered.application;
```
Comment (msm1992) [inline product-scenarios/4-application-development-using-apis/4.1-application-developers-should-be-able-to-use-apis-in-their-applications/4.1.3-delete-registered-application/src/test/java/org/wso2/am/scenario/tests/delete/registered/application/DeleteRegisteredApplicationNegativeTestCase.java]: Shall we add license?

File: product-scenarios/4-application-development-using-apis/4.1-application-developers-should-be-able-to-use-apis-in-their-applications/4.1.3-delete-registered-application/src/test/java/org/wso2/am/scenario/tests/delete/registered/application/DeleteRegisteredApplicationTestCase.java (hunk 4 above)
Comment (ruwiniwj) [reply product-scenarios/4-application-development-using-apis/4.1-application-developers-should-be-able-to-use-apis-in-their-applications/4.1.3-delete-registered-application/src/test/java/org/wso2/am/scenario/tests/delete/registered/application/DeleteRegisteredApplicationTestCase.java]: Fixed with https://github.com/wso2/product-apim/pull/5248

File: product-scenarios/4-application-development-using-apis/4.1-application-developers-should-be-able-to-use-apis-in-their-applications/4.1.3-delete-registered-application/src/test/java/org/wso2/am/scenario/tests/delete/registered/application/DeleteRegisteredApplicationNegativeTestCase.java (hunk 5 above)
Comment (ruwiniwj) [reply product-scenarios/4-application-development-using-apis/4.1-application-developers-should-be-able-to-use-apis-in-their-applications/4.1.3-delete-registered-application/src/test/java/org/wso2/am/scenario/tests/delete/registered/application/DeleteRegisteredApplicationNegativeTestCase.java]: Fixed with https://github.com/wso2/product-apim/pull/5248

File: product-scenarios/4-application-development-using-apis/4.1-application-developers-should-be-able-to-use-apis-in-their-applications/4.1.3-delete-registered-application/pom.xml [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+        <dependency>
+            <groupId>org.jacoco</groupId>
+            <artifactId>org.jacoco.agent</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>com.icegreen</groupId>
+            <artifactId>greenmail</artifactId>
+            <scope>test</scope>
+        </dependency>
+
```
Comment (VirajSalaka) [inline product-scenarios/4-application-development-using-apis/4.1-application-developers-should-be-able-to-use-apis-in-their-applications/4.1.3-delete-registered-application/pom.xml]: Additional line

File: product-scenarios/4-application-development-using-apis/4.1-application-developers-should-be-able-to-use-apis-in-their-applications/4.1.3-delete-registered-application/pom.xml [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+    </dependencies>
+
```
Comment (VirajSalaka) [inline product-scenarios/4-application-development-using-apis/4.1-application-developers-should-be-able-to-use-apis-in-their-applications/4.1.3-delete-registered-application/pom.xml]: additional line, can be removed

File: product-scenarios/4-application-development-using-apis/4.1-application-developers-should-be-able-to-use-apis-in-their-applications/4.1.3-delete-registered-application/pom.xml (hunk 6 above)
Comment (ruwiniwj) [reply product-scenarios/4-application-development-using-apis/4.1-application-developers-should-be-able-to-use-apis-in-their-applications/4.1.3-delete-registered-application/pom.xml]: Fixed with #5248

File: product-scenarios/4-application-development-using-apis/4.1-application-developers-should-be-able-to-use-apis-in-their-applications/4.1.3-delete-registered-application/pom.xml (hunk 7 above)
Comment (ruwiniwj) [reply product-scenarios/4-application-development-using-apis/4.1-application-developers-should-be-able-to-use-apis-in-their-applications/4.1.3-delete-registered-application/pom.xml]: Fixed with #5248

**Pull request merged**